scan-command/src/main/java/io/ballerina/scan/utils/ScanTomlFile.java

```diff
@@ -0,0 +1,112 @@
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan.utils;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * {@code ScanTomlFile} contains an in-memory representation of the scan.toml file.
+ *
+ * @since 0.1.0
+ * */
+public class ScanTomlFile {
+    private final Set<Platform> platforms;
+    private final Set<Analyzer> analyzers;
+    private final Set<RuleToFilter> rulesToInclude;
+    private final Set<RuleToFilter> rulesToExclude;
+
+    ScanTomlFile() {
+        this.platforms = new LinkedHashSet<>();
+        this.analyzers = new LinkedHashSet<>();
+        this.rulesToInclude = new LinkedHashSet<>();
+        this.rulesToExclude = new LinkedHashSet<>();
+    }
+
+    void setPlatform(Platform platform) {
+        platforms.add(platform);
+    }
+
+    void setAnalyzer(Analyzer analyzer) {
+        analyzers.add(analyzer);
+    }
+
+    void setRuleToInclude(RuleToFilter rule) {
+        rulesToInclude.add(rule);
+    }
+
+    void setRuleToExclude(RuleToFilter rule) {
+        rulesToExclude.add(rule);
+    }
+
+    /**
+     * Returns an unmodifiable {@link Set} of platforms.
+     *
+     * @return an unmodifiable set of platforms
+     */
+    public Set<Platform> getPlatforms() {
+        return Collections.unmodifiableSet(platforms);
+    }
+
+    /**
+     * Returns an unmodifiable {@link Set} of analyzers.
+     *
+     * @return an unmodifiable set of analyzers
+     */
+    public Set<Analyzer> getAnalyzers() {
+        return Collections.unmodifiableSet(analyzers);
+    }
+
+    /**
+     * Returns an unmodifiable {@link Set} of rules to include.
+     *
+     * @return an unmodifiable set of rules to include
+     */
+    public Set<RuleToFilter> getRulesToInclude() {
+        return Collections.unmodifiableSet(rulesToInclude);
+    }
+
+    /**
+     * Returns an unmodifiable {@link Set} of rules to exclude.
+     *
+     * @return an unmodifiable set of rules to exclude
+     */
+    public Set<RuleToFilter> getRulesToExclude() {
+        return Collections.unmodifiableSet(rulesToExclude);
+    }
+
+    public record Platform(String name, String path, Map<String, Object> arguments) {
+        /**
+         * @param name      in-memory representation of platform name
+         * @param path      in-memory representation of platform JAR path
+         * @param arguments in-memory representation of platform arguments
+         */
+        public Platform(String name, String path, Map<String, Object> arguments) {
+            this.name = name;
+            this.path = path;
+            this.arguments = Collections.unmodifiableMap(arguments);
+        }
```

Comment:
Since all you need is to make arguments unmodifiable (ie. other parameters are un-touched) you can just use the compact constructor
```suggestion
        public Platform {
            arguments = Collections.unmodifiableMap(arguments);
        }
```

Reply:
Also typically we add the doc comment at the record definition level not at the constructor [see](https://github.com/ballerina-platform/ballerina-lang/blob/0d2fd413d5425689735d259c403b7fdbf8e8846e/semtypes/src/main/java/io/ballerina/types/FixedLengthArray.java#L24)